packages/grant-explorer/src/features/round/ViewRoundPage.tsx

```diff
@@ -24,19 +25,59 @@ export default function ViewRound() {
   const { round, isLoading } = useRoundById(chainId!, roundId!);
 
   const [roundExists, setRoundExists] = useState(true);
+
+  const [applicationsOpen, setApplicationsOpen] = useState(false);
+
+  const [roundOpen, setRoundOpen] = useState(false);
+
   useEffect(() => {
+    const currentTime = new Date();
     setRoundExists(!!round);
+    if (round) {
+      if (
+        round.applicationsStartTime <= currentTime &&
+        round.applicationsEndTime >= currentTime
+      ) {
+        setApplicationsOpen(true);
+      }
+      if (
+        round.roundStartTime <= currentTime &&
+        round.roundEndTime >= currentTime
+      ) {
+        setRoundOpen(true);
+      }
+    }
```

Comment:
^ @bhargavaparoksham you recoken we might have more than 2 states ? 
Cause i think we have
- round not started
- applications open 
- applications ended
- round started
- round ended

^ these are possible states which the ticket doesn't account for but i think it might be relevant 
You reckoen we keep one `setRoundState` or have an individual state mgmt for each round state

Reply:
@thelostone-mc implemented this in a different way now. Please review it & let me know if yo have further feedback.